### PR TITLE
(Archive) Get archive entries by archival/retrieval id

### PR DESCRIPTION
**Pull request by islean**
## Description

This PR adds functionality for fetching entries from the Archive table providing either an archiving_job_id or a retrieval_task_id.

### Added

- Method get_archives.

### How to prepare for test

- [ ] ssh to hasta (depending on type of change)
- [ ] install on stage:
```bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t housekeeper -b get-archive-entries```

### How to test
- [ ] login to ...
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] code approved by
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: housekeeper/store/api/handlers/read.py [hunk 1]
```diff
@@ -249,3 +243,33 @@ def get_all_non_archived_files(self, tag_names: list[str]) -> list[File]:
             tag_names=tag_names,
             is_archived=False,
         ).all()
+
+    def get_archives(
+        self, archival_task_id: int = None, retrieval_task_id: int = None
+    ) -> Optional[list[File]]:
```
Comment (islean) [inline housekeeper/store/api/handlers/read.py:249]: Would we want support for lists of ids?

File: housekeeper/store/api/handlers/read.py [hunk 2]
```diff
@@ -249,3 +243,33 @@ def get_all_non_archived_files(self, tag_names: list[str]) -> list[File]:
...
+                task_id=retrieval_task_id,
+            ).all()
+        if archival_task_id:
+            return apply_archive_filter(
+                archives=self._get_query(table=Archive),
+                filter_functions=[ArchiveFilter.FILTER_BY_ARCHIVING_TASK_ID],
+                task_id=archival_task_id,
+            ).all()
+
+        else:
```
Comment (henrikstranneheim) [inline housekeeper/store/api/handlers/read.py:270]:
```suggestion
```
I do not think it is needed sice you return in all ifs

File: tests/conftest.py [hunk 3]
```diff
@@ -78,6 +78,12 @@ def archiving_task_id() -> int:
     return 1234
 
 
+@pytest.fixture(scope="function")
```
Comment (henrikstranneheim) [inline tests/conftest.py:81]:
```suggestion
@pytest.fixture(scope="session")
```

File: tests/conftest.py [hunk 4]
```diff
@@ -126,6 +132,12 @@ def spring_file_2_with_tags(sample_id: str, spring_tag: str, spring_file_2: Path
     return {"tags": [sample_id, spring_tag], "file": spring_file_2}
 
 
+@pytest.fixture(scope="function")
```
Comment (henrikstranneheim) [inline tests/conftest.py:135]:
```suggestion
@pytest.fixture(scope="session")
```

File: tests/conftest.py [hunk 5]
```diff
@@ -408,6 +421,12 @@ def spring_file_2(sequencing_files_dir: Path) -> Path:
     return Path(sequencing_files_dir, "lane2.spring")
 
 
+@pytest.fixture(scope="function")
```
Comment (henrikstranneheim) [inline tests/conftest.py:424]:
```suggestion
@pytest.fixture(scope="session")
```

File: tests/store/handlers/test_readhandler.py [hunk 6]
```diff
@@ -198,3 +198,79 @@ def test_get_ongoing_retrieval_tasks(
...
+def test_get_archive_entries_by_retrieval_id(retrieval_task_id: int, populated_store: Store):
+    """Tests that get_archives returns archive entries with the given archiving task id."""
+
+    # GIVEN a populated store with an archive entry
+
+    # WHEN getting entries with the matching retrieval task id
+    archives: list[Archive] = populated_store.get_archives(retrieval_task_id=retrieval_task_id)
+
+    # THEN an Archive entry should be returned
+    assert len(archives) > 0
```
Comment (henrikstranneheim) [inline tests/store/handlers/test_readhandler.py:226]:
```suggestion
    assert archives
```

File: tests/store/handlers/test_readhandler.py [hunk 7]
```diff
@@ -198,3 +198,79 @@ def test_get_ongoing_retrieval_tasks(
...
+    # WHEN fetching all archives
+    all_archives: list[Archive] = populated_store.get_archives()
+
+    # WHEN fetching all archives with a specified archiving task id
+    archiving_archives: list[Archive] = populated_store.get_archives(
+        archival_task_id=archiving_task_id
+    )
+
+    # THEN only archives with the matching retrieval task id is returned
+
```
Comment (henrikstranneheim) [inline tests/store/handlers/test_readhandler.py:247]:
```suggestion
```

File: tests/store/handlers/test_readhandler.py [hunk 8]
```diff
@@ -198,3 +198,79 @@ def test_get_ongoing_retrieval_tasks(
...
+def test_get_archive_entries_by_archiving_id(
+    archive: Archive, archiving_task_id: int, populated_store: Store
+):
+    """Tests that get_archives returns archive entries with the given archiving task id."""
+
+    # GIVEN a populated store with an archive entry
+
+    # WHEN getting entries with the matching archiving task id
+    archives: list[Archive] = populated_store.get_archives(archival_task_id=archiving_task_id)
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply tests/store/handlers/test_readhandler.py:211]: OR perhaps even better - start using According to [PEP 604](https://peps.python.org/pep-0604/#proposal) Is preferred to use Tag | None as @diitaz93 pointed out.

File: housekeeper/store/api/handlers/read.py (hunk 2 above)
Comment (henrikstranneheim) [reply housekeeper/store/api/handlers/read.py:270]: Yes - I just mean that if you will execute this block with or without the else if the execution got this far, but maybe I am missing something.

File: tests/store/handlers/test_readhandler.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (islean) [reply tests/store/handlers/test_readhandler.py:211]: Used the PEP 604 one now!

File: housekeeper/store/api/handlers/read.py (hunk 2 above)
Comment (islean) [reply housekeeper/store/api/handlers/read.py:270]: Oh! Now I see what you mean! Good catch!

**Comment by islean** [comment]
Tested deploying and making sure the CLI is responsive. Since this is no live functionality, this should be the necessary tests performed.

```
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/isak.ohlsson
done.
Log deploy... done.
housekeeper, version 4.8.21
θ64° 1d [isak.ohlsson@hasta:~] [S_base] 1m25s $ housekeeper --help
Usage: housekeeper [OPTIONS] COMMAND [ARGS]...

  Housekeeper - Access your files!
```

**Comment by islean** [comment]
Merged to master by IO

**Pull request merged**